File: p740p758XML/p740XML/Person.m

#import "Person.h"


@implementation Person
@synthesize lastName, firstName;

- (void)dealloc {
    [firstName release];
    [lastName release];
    [super dealloc];
}

- (void)encodeWithCoder:(NSCoder *)encoder {
    //[super encodeWithCoder: encoder];
    [encoder encodeObject:self->lastName forKey:@"last"];
    [encoder encodeObject:self->firstName forKey:@"first"];
}

- (id) initWithCoder:(NSCoder *)decoder {
    //self = [super initWithCoder: decoder];
    self = [super init];
    self->lastName = [decoder decodeObjectForKey:@"last"];
    [self->lastName retain];
    self->firstName = [decoder decodeObjectForKey:@"first"];
    [self->firstName retain];
    return self;
}

- (NSString*) description {
    return [NSString stringWithFormat: @"%@ %@", firstName, lastName];
}

@end

File: bk2ch23p810DocumentBrowser/DocumentBrowser/Person.swift


import UIKit

class Person: NSObject, Codable {
    var firstName : String
    var lastName : String
    
    override var description : String {
        return self.firstName + " " + self.lastName
    }
    
    init(firstName:String, lastName:String) {
        self.firstName = firstName
        self.lastName = lastName
        super.init()
    }
    
}

